pages/compare.js

```diff
@@ -1,26 +1,227 @@
+      newHidden.push(id);
+    }
+    this.setState({ hidden: newHidden });
+  }
+
+  /* Get an accordion section list with the components */
+  getList(list) {
+    const { hidden } = this.state;
+
+    return list.map((l, i) => (
+      <div className={`accordion-item ${hidden.includes(l.id) ? '-hidden' : ''}`} id={l.id} key={i}>
+        {l.el}
+      </div>
+    ));
+  }
+
+  /** Maps methods */
+  /* Config map params and set them in the map */
+  setMapParams(params) {
+    Object.keys(params).forEach((key) => {
+      const mapParams = {
+        zoom: +params[key].zoom || MAP_OPTIONS.zoom,
+        center: {
+          lat: +params[key].lat || MAP_OPTIONS.center[0],
+          lng: +params[key].lng || MAP_OPTIONS.center[1]
+        }
+      };
+      this.props.setSingleMapParamsFromUrl(mapParams, key);
+    });
+  }
+
+  /* Map config */
+  updateMap(map, url, key) {
+    this.props.setSingleMapParams(
+      {
+        zoom: map.getZoom(),
+        center: map.getCenter(),
+        key
+      }, url, key);
+  }
+
+  /* Creat all maps with their own properties */
+  getAreaMaps(layers) {
+    const { mapState, indicators } = this.props;
+
+    return Object.keys(mapState.areas).map((key) => {
+      const listeners = {
+        moveend: (map) => {
+          this.updateMap(map, this.url, key);
+        }
+      };
+
+      const mapMethods = {
+        attribution: '&copy; <a href="http://www.openstreetmap.org/copyright" target="_blank">OpenStreetMap</a>',
+        tileLayers: [
+          { url: process.env.BASEMAP_TILE_URL, zIndex: 0 },
+          { url: process.env.BASEMAP_LABEL_URL, zIndex: 10000 }
+        ]
+      };
+
+      const mapOptions = {
+        zoom: mapState.areas[key].zoom,
+        minZoom: MAP_OPTIONS.minZoom,
+        maxZoom: MAP_OPTIONS.maxZoom,
+        zoomControl: MAP_OPTIONS.zoomControl,
+        center: [mapState.areas[key].center.lat, mapState.areas[key].center.lng]
+      };
+
+      return {
+        id: key,
+        el: (
+          <Map
+            mapOptions={mapOptions}
+            mapMethods={mapMethods}
+            listeners={listeners}
+            layers={layers}
+            indicatorsLayersActive={indicators.layersActive}
+            markers={[]}
+            markerIcon={{}}
+          />
+        )
+      };
+    });
+  }
+
+  /* Add area */
+  onAddArea() {
+    Object.keys(this.props.mapState.areas).length < 3 &&
+      this.props.addArea();
+  }
+
+  /* Reamove area */
+  onRemoveArea(id) {
+    Object.keys(this.props.mapState.areas).length > 1 &&
+      this.props.removeArea(id);
+  }
+
   render() {
-    const { url, session } = this.props;
+    const { session } = this.props;
+    const { url, mapState, indicators } = this.props;
+    const layers = setLayersZIndex(indicators.layers, indicators.layersActive);
+    const areaMaps = this.getAreaMaps(layers);
+
+    // Widget samples
+    const indicatorsWidgets = Object.keys(mapState.areas).map(key => (
```

Comment:
Too many things are happening on this component/page. I think that we should create the component Area